app/src/main/java/com/example/sharewhatyoucanproject/login/LoginActivity.kt

```diff
@@ -0,0 +1,110 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.app.ProgressDialog
+import android.content.Intent
+import android.os.Bundle
+import android.provider.Settings
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.DrawerActivity
+import com.example.sharewhatyoucanproject.SelectUserActivity
+import com.example.sharewhatyoucanproject.databinding.ActivityLoginBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.editSharedPreferencesSelector
+import com.example.sharewhatyoucanproject.utils.showToast
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    lateinit var loginViewModel: LoginViewModel
+    lateinit var progressDialog: ProgressDialog
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        progressDialog = ProgressDialog(this)
+        progressDialog.setTitle("Please Wait")
+        supportActionBar?.hide()
+
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(),
+        )[LoginViewModel::class.java]
+
+        val type = intent.getIntExtra("type", 0)
```

Comment:
it would be good to add this to your utility class so you only have to write this logic once, and then it can also be tested:

```
fun getUserType(intent: Intent): UserType {
    val type = intent.getIntExtra("type", 0)
    return UserType.values()[type]
}
```

usage:
```
val type = getUserType(intent)
```